### PR TITLE
The Grouse Outdoors

### PR DESCRIPTION
**Pull request by AndrewL97**
This PR adds outdoor effects to MS in the form of Weather, using particle effects, and Sunlight, which adds to the lighting system.

Each turf is given an Outdoor_effects object, that holds up to two overlays - one for weather and one for sunlight.

Sunlight is governed by three "states" on a turf, based on a roof_type 
SKY_BLOCKED
SKY_VISIBLE
SKY_VISIBLE_BORDER

Rather than setting colours on individual overlays for each turf - we use a greyscale overlay, and a coloured fullscreen object to colour everything. This means that we can animate sunlight changes, without completely melting the server.

Weather is affected by whether or not the roof above is weatherproof - for now this is limited to space/open_space

### CODE REVIEW

**Comment by AndrewL97** [comment]
Lint issues regarding particle effects are lies 
fadein is a native byond var for /particles
and generators can be chained together

File: mojave/modules/outdoor_effects/code/modules/particool/particool.dm [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+	//Normal Matrixes allow 6
+	switch(length(L))
+		if(6)
+			return matrix(L[1],L[2],L[3],L[4],L[5],L[6])
+
+/datum/particle_editor/proc/stringToList(str, toNum = FALSE)
+	. = splittext(str,regex(@"(?<!\\),"))
+	if(toNum)
+		for(var/i = 1; i <= length(.); ++i)
```
Comment (Koshenko) [inline mojave/modules/outdoor_effects/code/modules/particool/particool.dm:57]: Could probably optimize this slightly with `for(i in list)

File: mojave/modules/outdoor_effects/code/_onclick/hud/fullscreen.dm [hunk 2]
```diff
@@ -0,0 +1,21 @@
+/* Our sunlight planemaster mashes all of our sunlight overlays together into one             */
+/* The fullscreen then grabs the plane_master with a layer filter, and colours it             */
+/* We do this so the sunlight fullscreen acts as a big lighting object, in our lighting plane */
+/atom/movable/screen/fullscreen/lighting_backdrop/Sunlight
```
Comment (Koshenko) [inline mojave/modules/outdoor_effects/code/_onclick/hud/fullscreen.dm:4]: Lowercase

File: mojave/modules/outdoor_effects/code/controllers/subsystem/outdoor_effects.dm [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+	var/color = ""
+	var/start = 216000 // 6:00 am
+
+
+// /datum/time_of_day/Midnight_am
+//  name = "Midnight AM"
+//  color = "#050d29"
+//  start = "0" //12:00:00 AM
+
+/datum/time_of_day/Dawn
```
Comment (Koshenko) [inline mojave/modules/outdoor_effects/code/controllers/subsystem/outdoor_effects.dm]: Lowercase all the datums in here

File: mojave/modules/outdoor_effects/code/controllers/subsystem/particle_weather.dm [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+		weatherEffect.filters += filter(type="alpha", render_source=WEATHER_RENDER_TARGET)
+		weatherEffect.mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	return weatherEffect
+
+/datum/controller/subsystem/ParticleWeather/proc/SetparticleEffect(particles/P)
+	particleEffect = P
+	weatherEffect.particles = particleEffect
+
+/datum/controller/subsystem/ParticleWeather/proc/stopWeather()
+	qdel(runningWeather)
```
Comment (Koshenko) [inline mojave/modules/outdoor_effects/code/controllers/subsystem/particle_weather.dm]: `QDELL_NULL()` is a nice shorthand macro for `qdel` then set the reference to null.

**Review by Koshenko — CHANGES_REQUESTED** [review]
Added a few small requests for changes.

**Pull request merged**